### ISSUE TITLE
isLong method is working wrong for max lengthed long values.

### ISSUE THREAD

**Issue by kubudi** [advantageous/boon#23]
TLDR: Json parsing is failing if long values has same digit count with `Long.MAX_VALUE`. 

Details:
I'm using gatling for performance testing, and it's using boon under the hood. When I tried to parse a specific json I got an interesting exception:
```java
io.advantageous.boon.core.Exceptions$SoftenedException: not a long
	at io.advantageous.boon.core.Exceptions.die(Exceptions.java:97)
	at io.advantageous.boon.core.value.NumberValue.longValue(NumberValue.java:231)
	at io.advantageous.boon.core.value.NumberValue.doToValue(NumberValue.java:130)
	at io.advantageous.boon.core.value.NumberValue.toValue(NumberValue.java:104)
	at io.advantageous.boon.core.value.LazyValueMap.buildMap(LazyValueMap.java:226)
	at io.advantageous.boon.core.value.LazyValueMap.entrySet(LazyValueMap.java:214)
...
```
I had nothing to do with any long values, and this code was working fine with other jsons. 
Funny thing is, it was sometimes successful with no changes whatsoever. So I dig 

As the exception implies, for parsing Long values, the following method is used under `NumberValue.java`:
```java
    @Override
    public long longValue () {
        if( CharScanner.isLong(buffer, startIndex, endIndex - startIndex)){
            return CharScanner.parseLong(buffer, startIndex, endIndex);
        } else {
            return Exceptions.die(int.class, "not a long");
        }
    }
```
This is where I get the exception from.

Under `CharScanner.java` `isLong` method is comparing given value to `Long.MAX_VALUE` by their length:
```java
 public static boolean isLong( char[] digitChars, int offset, int len ) {
        String cmpStr = digitChars[offset]=='-' ? MIN_LONG_STR_NO_SIGN : MAX_LONG_STR;
        int cmpLen = cmpStr.length();

        if ( len > cmpLen ) return false;


        if (verifyValueFitsInNumber(digitChars, offset, len, cmpStr, cmpLen)) return false;

        if (digitChars[offset]=='-') {
            return isDigits(digitChars, ++offset, --len);
        } else {
            return isDigits(digitChars, offset, len);
        }

    }
```
Length of `Long.MAX_VALUE`(9223372036854775807) is 19.
This comparison is causing Json parsing(probably it has some other impacts too) to fail.

Later I realized successfull cases was because, in these scenarios, long field in my json was returning a smaller number. Mystery solved.

I didn't investigated that much but I believe a similar problem is present for integer as well.

Sorry for not coming with a fix, I'm a bit busy atm. But I thought the sooner I raise this the better 

Note: 
I guess when throwing this exception we should use `long.class` instead. 
`return Exceptions.die(int.class, "not a long");`

**Comment by kubudi** [comment]
Ah, sorry for not checking. #354 resolves this.
